src/Runner.jl

```diff
@@ -662,6 +663,21 @@ function generate_compiler_wrappers!(platform::AbstractPlatform; bin_path::Abstr
             write_wrapper(rustc, p, "$(t)-rustc")
             write_wrapper(rustup, p, "$(t)-rustup")
             write_wrapper(cargo, p, "$(t)-cargo")
+
+            # For FreeBSD and macOS we need to create an unversioned link for
+            # gcc because that's the linker our Rust toolchain expects:
+            # https://github.com/JuliaPackaging/Yggdrasil/blob/fff0583bc2d8f32e450c427684f295524f38535d/0_RootFS/Rust/build_tarballs.jl#L115-L126.
+            if Sys.isbsd(p) && os_version(p) !== nothing
+                delete!(tags(p), "os_version")
```

Comment:
I'd feel more comfortable if we created a new `p` object here instead of mutating the existing one; we may want to use `p` in the future, and it would be bad if we didn't have the `os_version` tag when we expected to have it.

Reply:
Fair enough!